lib/ansible/modules/cloud/ovirt/ovirt_templates.py

```diff
@@ -326,32 +352,33 @@ def main():
             # Find the the unregistered Template we want to register:
             templates = templates_service.list(unregistered=True)
             template = next(
-                (t for t in templates if t.name == module.params['name']),
+                (t for t in templates if (t.id == module.params['id'] or t.name == module.params['name'])),
                 None
             )
             changed = False
             if template is None:
-                # Test if template is registered:
-                template = templates_module.search_entity()
+                template = search_entity()
                 if template is None:
                     raise ValueError(
-                        "Template with name '%s' wasn't found." % module.params['name']
+                        "Template '%s(%s)' wasn't found." % (module.params['name'], module.params['id'])
                     )
             else:
+                # Register the template into the system:
                 changed = True
                 template_service = templates_service.template_service(template.id)
-                # Register the template into the system:
                 template_service.register(
+                    allow_partial_import=module.params['allow_partial_import']
+                    if module.params['allow_partial_import'] else None,
```

Comment:
No need for `if` here, because the default is `None` anyway and if `False`, `None` is passed, which is wrong.